File: app/src/main/java/szeptunm/corner/dataaccess/database/DatabaseTransaction.java
package szeptunm.corner.dataaccess.database;

import io.reactivex.Completable;
import javax.inject.Inject;

public class DatabaseTransaction {

    @Inject
    CornerDatabase database;

    @Inject
    DatabaseTransaction() {
    }

    public Completable runTransaction(Runnable runnable) {
        return Completable.fromAction(() -> database.runInTransaction(runnable));
    }
}

File: app/src/main/java/szeptunm/corner/dataaccess/database/DatabaseTransaction.kt
package szeptunm.corner.dataaccess.database

import io.reactivex.Completable
import javax.inject.Inject

class DatabaseTransaction @Inject constructor(var database: CornerDatabase) {

    fun runTransaction(runnable: Runnable): Completable {
        return Completable.fromAction { database.runInTransaction(runnable) }
    }
}